doc/ug/install_instructions/index.md

```diff
@@ -4,30 +4,89 @@ title: Install Build Requirements
-* Physical access to that machine, root permissions and a graphical environment.
-* Python 3.5.2 or newer. Python 3.6+ is recommended.
-* A C++14 capable compiler. GCC 5 or Clang 3.5 should meet this requirement.
-* 60 GB or more of disk space.
-  EDA tools like Xilinx Vivado can easily take up 40 GB each.
-* We develop and test on the following Linux distributions:
-  * Ubuntu 16.04 LTS
-  * Debian testing
-  * openSUSE Tumbleweed
-  * TODO: Check RHEL/CentOS and SLES (used in many commercial environments)
-
-TODO: Be more specific about the system requirements, especially the Linux distribution.
+* Physical access to that machine, root permissions and a graphical environment are assumed in this guide.
+  However, with an appropriate setup, none of these things are strictly needed.
+* 60 GB or more of disk space, depending on the EDA tools used.
+  (EDA tools like Xilinx Vivado can easily take up 40 GB each.)
+* Linux
+  * Ubuntu 16.04 LTS is the recommended reference platform.
+    Our continuous integration setup runs on Ubuntu 16.04 LTS, which gives us the most confidence that this distribution works out of the box.
+  * We do our best to support other Linux distributions.
+    However, we cannot guarantee they can be used "out of the box" and might require updates of packages.
+    Please file a [GitHub issue](https://github.com/lowRISC/opentitan/issues) if you need help or would like to propose a change to increase compatibility with other distributions.
+    * RedHat Enterprise Linux (RHEL)/CentOS.
+
+      On RHEL/CentOS not all required packages are part of the base distribution; however, they can be installed easily.
+      All required packages except for a C/C++ compiler can be obtained by enabling the [EPEL repository](https://fedoraproject.org/wiki/EPEL).
+      A sufficiently recent version of GCC or LLVM (clang) can be obtained through [Red Hat Software Collections](https://www.softwarecollections.org/en/).
+      SCLs can also be used to obtain newer version of Python if needed.
+
+    * Many of our contributors use rolling release distributions, which generally include the latest stable releases of all software packages, such as Debian unstable, openSUSE Tumbleweed, or Arch Linux.
+
+The following software packages are required to build and use the software, hardware, and tools in the OpenTitan repository.
+Not every interaction with the repository requires all tools -- however, we recommend users installing all dependencies listed below for a seamless development experience.
+
+Note: Version numbers given below indicate a minimum version, newer versions are generally expected to work, unless explicitly stated.
+For packages listed below without a version number we have not determined a minimum version.
+
+* git
+* Python 3.5.2 with pip.
+  Additional Python dependencies are installed through pip.
+* A C++14 capable compiler.
+  GCC 5 or Clang 3.5 should meet this requirement.
+* clang-format
```

Comment:
Somewhere in the docs, we say exactly what version is required. Right now, I believe it's `3.8.x`. 

I think the "somewhere" is in the `utils/container/README.md`, which might need merging with this?

Reply:
We don't require a specific version at the moment, whatever is available as part of a supported LLVM version is fine. So effectively clang-format 3.5+ at the moment.

Reply:
Not true. More recent clang-format versions have subtle differences to the results of running clang-format in CI. See for instance #2035 and #2082. This is an issue googlers have run into, which is why a specific version of clang-format was documented at all.

Reply:
Can you point me to this documentation?

Yes, we use a specific version in CI and using another version will be painful on occasion, but I don't think we have a policy requiring a specific version at this point.

Reply:
I added a sentence on clang-format 3.8 being recommended, please let me know if that addresses your concerns @lenary.